src/main/scala/abalone/variant/Variant.scala

```diff
@@ -102,6 +103,144 @@ abstract class Variant private[variant] (
         )
     }.toMap
 
+  def validMovesOf2(situation: Situation, validMoves1: Map[Pos, List[Move]]): Map[Pos, List[(String, Move)]] =
+    situation.board.piecesOf(situation.player).flatMap {
+      case ((pos, piece)) => {
+        Map(
+          pos -> pos.neighbours.flatMap {
+            case Some(neighbour) if(situation.board.piecesOf(situation.player).contains(neighbour)) => {
+              neighbour.neighbours.flatMap {
```

Comment:
it would be better if we could just get the neighbour of `neighbour` only in the direction of pos->neighbour? all of the other neighbour of neighbours are just irrelevant?

Reply:
so I think we want to consider the positions where ab could land, displayed as uppercase letter in the situation below
```
   *  *  A  B  *
 *  *  a  b  C  *
*  *  *  E  D  *  *  
```
?

Reply:
But ab can only push into C right? theres no valid move ever for ab to interact with A, B, D or E, so why even call a function that returns A,B,C,D,E?

Reply:
In Abalone there are 3 kinds of moves.
  Let's use the top 2 rows of a board to illustrate (numbers are empty squares, Z is an opponent marble)

        Z a b c 7
       1 2 3 4 5 6
    
    - line moves are marbles moving in line to an empty square :
       'a' to '7' would move three marbles to the right. This is the same as if 'a' jumped over 'b' and 'c' to land on '7'.
       'c' to '7' would move one marble to the right.
    - pushes are line moves targeting a square hosting an opponent marble
       They will be processed as two line moves (one jump per player).
       'c' to 'Z' and 'b' to 'Z' are pushes to the left
    - side moves can only be described starting from the right origin (figure out the longest diagonal) :
       'a' to '5' is a downRight side move of three marbles.
       'c' to '2' is a downLeft side move of three marbles.
       'a' to '4' is a downRight side move of two marbles (only 'a' and 'b' would move).
       'c' to '3' is a downLeft side move of two marbles (only 'c' and 'b' would move)
       'a' to '3' or 'c' to '4' are line moves of a single marble